src/proto/message.proto

```diff
@@ -163,11 +163,47 @@ message HiveMindHostApiResponse {
 
 // HiveConnectHivemind API
 // Used by the HiveMind with the HiveConnect, this is the API that the HC expose to the HM
+message HiveConnectNetworkAccess {
+    string ssid = 1;
+    string password = 2;
+}
+
+message HiveConnectRootNode {
+    bool is_root = 1;
+}
+
+message HiveConnectMeshEnable {
+    bool use_mesh = 1;
+}
+
+message HiveConnectNetworkConfigSetRequest {
+    HiveConnectNetworkAccess network_access = 1;
+    HiveConnectRootNode root_node = 2;
+    HiveConnectMeshEnable mesh_enable = 3;
+}
```

Comment:
Is there a reason to have them all in individual messages instead of the fields directly in the config set?

Reply:
Exactly, and also, to have them more modular if changes need to be made in the future